src/js/common/components/Modal.jsx

```diff
@@ -78,7 +78,7 @@ class Modal extends React.Component {
     }
 
     return (
-      <div className={modalCss} id={id} role="alertdialog" aria-labelledby={`${id}-title`}>
+      <div className={modalCss} id={id} role="alertdialog" aria-labelledby={id && title ? `${id}-title` : undefined}>
```

Comment:
I think it needs an id if you include a title, but otherwise I don't think we would need to force all modals to have an id. I think the id is a new-ish addition to the component.